metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -171,20 +209,22 @@ class MetalsLanguageServer(
   private val savedFiles = new ActiveFiles(time)
   private val recentlyOpenedFiles = new ActiveFiles(time)
   private val recentlyFocusedFiles = new ActiveFiles(time)
-  private val languageClient = new DelegatingLanguageClient(NoopLanguageClient)
   val isImportInProcess = new AtomicBoolean(false)
-  @volatile
-  var userConfig: UserConfiguration = UserConfiguration()
   var excludedPackageHandler: ExcludedPackagesHandler =
     ExcludedPackagesHandler.default
 
-  var ammonite: Ammonite = _
   private val mainBuildTargetsData = new TargetData
-  val buildTargets: BuildTargets = new BuildTargets()
-  buildTargets.addData(mainBuildTargetsData)
+  val buildTargets: BuildTargets =
+    BuildTargets.from(workspace, mainBuildTargetsData, tables)
+
   private val buildTargetClasses =
     new BuildTargetClasses(buildTargets)
-  private var doctor: Doctor = _
+
+  private val sourceMapper = SourceMapper(
+    buildTargets,
+    buffers,
+    () => workspace,
```

Comment:
`() => workspace` can be changed to just `workspace`

Reply:
Just to be sure, this is because the workspace is already known right?

Reply:
yup